linkerd/io/src/boxed.rs

```diff
@@ -86,6 +87,38 @@ impl Io for BoxedIo {
     }
 }
 
+impl From<TcpStream> for BoxedIo {
+    fn from(stream: TcpStream) -> Self {
+        Self::new(stream)
+    }
+}
+
+impl<S: Io + Unpin + 'static> From<tokio_rustls::server::TlsStream<S>> for BoxedIo {
+    fn from(stream: tokio_rustls::server::TlsStream<S>) -> Self {
+        Self::new(stream)
+    }
+}
+
+impl<S: Io + Unpin + 'static> From<tokio_rustls::client::TlsStream<S>> for BoxedIo {
```

Comment:
Is it possible to implement this as just `From<I>`, since the TLS stream already implements Io? Or does the compiler get grumpy?

Reply:
`BoxedIo` itself implements `Io`, so it conflicts with the existing `From<Self>` impl that exists for everything. that was the first thing I tried.